src/components/find-on-shelf/FindOnShelfPeriodicalDropdowns.tsx

```diff
@@ -61,10 +69,14 @@ const FindOnShelfPeriodicalDropdown: FC<FindOnShelfPeriodicalDropdownProps> = ({
           <select
             className="dropdown__select"
             aria-label={t("findOnShelfModalPeriodicalEditionDropdownText")}
+            defaultValue={selectedPeriodical.volumeNumber}
             onChange={(e) =>
               setSelectedPeriodical({
-                selectedYear,
-                selectedEdition: e.target.value
+                volumeYear: selectedYear,
+                volumeNumber: e.target.value,
+                displayText: toBeSelectedPeriodical?.displayText || "",
+                itemNumber: toBeSelectedPeriodical?.itemNumber || "",
+                volume: toBeSelectedPeriodical?.volume || ""
               })
```

Comment:
Consider whether this can be replaced by `setSelectedPeriodical(toBeSelectedPeriodical)`?

It seems like all the props are transferred anyway.

Reply:
Good point, but that would be a bit of a problem because the types don't exactly match. The periodical edition type that we made doesn't 100% match the API spec as some of the items can be undefined there. I could give it a go though and edit our custom type.

Reply:
It seems like too much Typescript drama. It complaints about the changes from around the app. I'm sticking with this 🙈